packages/relay/src/lib/db/repositories/hbarLimiter/ethAddressHbarSpendingPlanRepository.ts

```diff
@@ -102,6 +102,21 @@ export class EthAddressHbarSpendingPlanRepository {
     }
   }
 
+  /**
+   * Deletes all spending plans associted with EVM addresses from cache.
+   *
+   * @param {RequestDetails} requestDetails - The request details for logging and tracking.
+   * @returns {Promise<void>} - A promise that resolves when all IP address spending plans are deleted.
+   */
+  async deleteAll(requestDetails: RequestDetails): Promise<void> {
```

Comment:
This feels like a method that could be in a base class that the inheriting class could call with an additional log

Reply:
@Nana-EC maybe i dont quite get what you mean. But if this method is specifically deleting entries in the collection for eth address, why should it be called from the base class and just add a log? Its kinda class specific.

Reply:
Looking at the plans the logic seems very similar. Feel like a shared method that can be scoped based on a passed in param

Reply:
https://github.com/hashgraph/hedera-json-rpc-relay/issues/3127 Opened a followup issue